### DISCUSSION TITLE
How do you interact with storage on Android

### DISCUSSION THREAD

**Discussion by yrezgui** [google/modernstorage#64] [General]
Hi everyone,
As a devrel on Android, sometimes I struggle to get insights on how developers interact with storage on Android.
Since Scoped Storage introduction, many APIs have changed and I want to know more details about your usage.
Please upvote using reactions and feel free to add comments to give more details:
- `java.io File` 👍
- `java.nio File` 🎉
- `MediaStore` ❤️
- `Storage Access Framework` (via `ContentResolver`) 🚀
- `NDK File` 👀

**Comment by ColtonIdle** [comment]
bad answer but... idk I just randomly copy code snippets on stackoverflow until it works. storage and files on android make my head hurt.

Common cases I've come across in my 5+ years as an android dev
1. save a file (like a pdf) to the users device, BUT save it outside of the "app sandbox" so that the pdf is still available even if they uninstall our app
2. save a file, like some json response or something, into my app sandbox in order to treat it like a cache
3. select a picture (for like a profile photo) and send it to our server
4. a user takes a picture with their camera app, but the photo is accessible in our app to send (similar to `3` ^)
So far, I think this library covers 1-3, but not 4. But 4 might just be out of scope for this lib I guess?

**Reply by yrezgui** [reply]
What's the difference between 2 & 4? You just save it in your internal storage, right?

**Reply by ColtonIdle** [reply]
Sorry. I guess I was moreso trying to say "I'm in my app. I press a "take picture" button that opens up the camera app, and then the camera app responds back to me with a URI (or something) and I'd like to read that photo and display it in my app"

**Comment by Tekkerue** [comment]
I use the LibGDX framework and its AndroidFileHandle class uses java.io.File. However, LibGDX's FileHandle functionality has been severely limited since the introduction of scoped storage and I haven't been able to find a solution to get it working the way it used to work. Someone on the LibGDX Discord pointed me here to ModernStorage and I'm hoping this will make file access much easier like it used to be. Using java.io.File might make it easier for the LibGDX developers to get the FileHandle working in the same way again.

**Comment by RobertAngermayr** [comment]
i user java.io File for my apps. 
At the moment i feel completely lost because i'm using binary files (and textfiles) and not mediafiles.
I'm 
downloading the binaries from the internet (automatically with my app)
recieve an email with the attached binary
create the files myself on the pc.
the app handles the access (no filepicker is used) in its own way.
create textfiles (kind of xml) to store on my phone, send them via email or messenger.

in the old (good) world i have an folder in root where all these files are stored.
In the new world i still dont know where to put them...
